documentation/production/tutorial-production-mode-basic.asciidoc

```diff
@@ -84,3 +84,102 @@ When minifying code, it is often obscured, which makes it harder to read, hence
 Bundling is an optimization where we merge multiple files to a single collection so that the browser doesn't need to
 request so many files from the server. 
 This makes the application load faster.
+
+== Checklist for the production mode app artifact
+
+Vaadin Servlet configures itself from the `flow-build-info.json` file, which is located in the `META-INF/VAADIN/config/` resource package.
+The resources location differ for different artifact types, for example:
+
+. Spring Boot jar file places resources into the `BOOT-INF/classes/` folder in the jar file;
+. WAR archive places resources into the `WEB-INF/classes/` folder.
+
+If the production mode has been activated properly, the contents of the `flow-build-info.json` should be as follows:
+
+.flow-build-info.json
+[source, json]
+----
+{
+  "compatibilityMode": false,
+  "productionMode": true,
+  "enableDevServer": false,
+  "chunks": {
+    "fallback": {
+      "jsModules": [
+        "@vaadin/vaadin-icons/vaadin-icons.js",
+        "@vaadin/vaadin-grid/src/vaadin-grid-tree-toggle.js",
+// etc etc
+        "frontend://ironListConnector.js"
+      ],
+      "cssImports": [
+      ]
+    }
+  }
+}
+----
+
+(Note that the `productionMode` property is set to `true`).
+
+It's very important to have this file on classpath. If it's not, Vaadin Servlet
+will then try to auto-configure itself and sometimes wrongly so. For example when launching
```

Comment:
I'm not sure I understand. Where does the `project.basedir` come from - perhaps this is a reference to Maven's `${project.basedir}`? However Maven properties are not accessible by Vaadin Servlet on runtime :thinking: Could you please provide more info on this?

Reply:
Yes by the base the project.basedir will be used as given from maven, but if that doesn't exist then the fallback comment now states:

```
     * Accept user.dir or cwd as a fallback only if the directory seems to be a
     * Maven or Gradle project. Check to avoid cluttering server directories
     * (see tickets #8249, #8403).
```

Reply:
Thank you. I wonder whether the `project.basedir` property has a merit. I can see that it can be set either via web.xml, or a `@WebServlet` annotation, or Spring application.properties, however neither of those three locations should contain hard-coded paths to a dev folder on one developer machine.

However, since this article only applies to production mode while `project.basedir` is only applied to development mode, I'd propose to avoid mentioning those properties at all in this particular article.

Reply:
The properties are not as such important, but this should instead of talking about auto configuring and creating things in wrong folders just note that the server tries to start up in development mode causing it to try to use different properties to find the development time files.

Reply:
Gotcha. I think it's a good idea to write a similar guide to troubleshoot a development mode, then link that guide from this one. However, I'll add a couple of remarks in this regard to this guide as well.

Reply:
I've mentioned the properties in the guide; since the text was extracted into a separate file, let us continue the discussion there.